### ISSUE TITLE
"forwardedfor = on" causing errors

### ISSUE THREAD

**Issue by Kenny690** [e2guardian/e2guardian#373]
Something wrong is going on.
After I turned on forwardedfor I'm getting a lot of errors

2018.4.6 14:02:36 10.20.25.244 10.20.25.244 https://jim46.mail.ru:443 DENIED Failed to nogotiate ssl connection to server CONNECT 0 0 SSL SITE 1 403 - - default - - - -

If I'm turning off sslmitm I'm not getting https sites opened at all, but still e2guardian writes code 200 in the log and squid has nothing from e2 in logs.

**Comment by philipianpearce** [comment]
Have fixed two bugs which relate to this in v5.0 and v5develop.

@Kenny690 I also had the same problem when testing the forwardedfor fix. Fixed it by allowing the forwarded ip range in squid.conf.  i.e. once you set folllow_x_forwarded_for in squid - squid tests the forwarded_for IPs in permission checking.    So in your set-up squid is rejecting the requests.

**Comment by Kenny690** [comment]
I'VE FOUND A PROBLEM! HA-HA!

So the problem is this lines in squid:

acl AuthorizedUsers proxy_auth REQUIRED
http_access allow AuthorizedUsers

after I removed them - forwardedfor is started to work. I have proxy-ntlm turned on, but seems like e2 can't work with it or I doing something wrong.

**Comment by philipianpearce** [comment]
Does the proxy_auth work if you disable the follow_forward_for in squid? With forwardedfor = on? = off? 

Regards 

Philip 

----- Original Message -----

**Comment by Kenny690** [comment]
I tried to comment all follow_x_forwarded_for rules and get strange things. Internet works, I see all requests in e2g's access.log, but no requests in squid.conf. And auth is not working.
Then I tried to change config a bit and got something new. 
I turned on authrequiresuserandgroup, turned forwarded for and follow_x_forwarded_for. In e2g log I have a lot of 407, but... I have the same in squid log. It recognise all machines, but can't authenticate them. If I connect to squid directly I can see my username, so squid's NTLM is working fine.
